src/components/FileTree/index.tsx

```diff
@@ -69,10 +70,10 @@ export class FileTreeBase extends React.Component<Props> {
   }
 
   _loadData = () => {
-    const { dispatch, tree, version } = this.props;
+    const { comparedToVersionId, dispatch, tree, version } = this.props;
 
     if (version && !tree) {
```

Comment:
What if you adjusted this if statement so that when `comparedToVersionId` does not match `state.fileTree.comparedToVersionId` it dispatches `buildTree()` again? Would that work without needing to do `forceReloadVersion`?

Reply:
> What if you adjusted this if statement so that when comparedToVersionId does not match state.fileTree.comparedToVersionId it dispatches buildTree() again?

In that way, `buildTree()` only take the old `version` object as a parameter, and change `fileTree.comparedToVersionId` to the new one, but no change on `fileTree.tree` at all.

I used `forceReloadVersion` flag to update `version` . `buildTree()` has to be dispatched after reloading, otherwise `fileTree.tree` will not be updated.

Reply:
Ah, right, we've run into this before. You could look at `FileTreeNode` to see a similar example. It accepts [`versionId` as the prop](https://github.com/mozilla/addons-code-manager/blob/941571bc6ed4d27e142e416ac4813897af611090/src/components/FileTreeNode/index.tsx#L36) instead of `version` then it [maps the latest `version` object from state](https://github.com/mozilla/addons-code-manager/blob/941571bc6ed4d27e142e416ac4813897af611090/src/components/FileTreeNode/index.tsx#L346). Even though that doesn't take `comparedToVersionId` into consideration, it *might* work since `DiffView` will always be loading the right `version`. Maybe try that and see?

Reply:
`FileTree` used the same way to map the `version` from state as `FileTreeNode`. The `version` object is always latest. But it seems a little tricky to dispatch `buildTree` properly, I tried something like (in `_loadData`)
```typescript
if (version && (!tree || fileTree.comparedToVersionId !== comparedToVersionId)) {
    dispatch(fileTreeActions.buildTree({ version }));
}
```
But `componentDidUpdate` will be called several times before `version` is updated, and `buildTree` is dispatched too early. And `buildTree` will not be dispatched again when `version` is updated since `fileTree.comparedToVersionid` is already updated ( `===comparedToVersionId`).

Reply:
> `FileTree` used the same way to map the `version` from state as `FileTreeNode`.

Oh, ha, I totally missed that. So it does.